apps/google-analytics-4/frontend/src/providers/SegmentAnalyticsProvider.tsx

```diff
@@ -0,0 +1,24 @@
+import { useSDK } from '@contentful/react-apps-toolkit';
+import { AnalyticsBrowser } from '@segment/analytics-next';
+import { config } from 'config';
+import { createContext, ReactNode } from 'react';
+
+export const SegmentAnalyticsContext = createContext<{ segmentAnalytics: AnalyticsBrowser | null }>(
+  { segmentAnalytics: null }
+);
+
+export const SegmentAnalyticsProvider = (props: { children: ReactNode }) => {
+  const sdk = useSDK();
+  const writeKey = config.segmentWriteKey;
+  const segmentAnalytics = AnalyticsBrowser.load({ writeKey });
+
+  // this is async, but per the Segment docs it doesn't need to be awaited since identify can safely be called
+  // before or after calls to page(), track() etc.
+  segmentAnalytics.identify(sdk.ids.user);
```

Comment:
double check that this doesn't need to fallback to `sdk.ids.app` when/if the user id context isn't set